src/Core/Managed/Shared/Channel/InMemoryChannel.cs

```diff
@@ -133,7 +143,14 @@ public void Flush()
         /// <param name="timeout">Timeout interval to abort sending.</param>
         public void Flush(TimeSpan timeout)
         {
-            this.transmitter.Flush(timeout);
+            if (!isDisposed)
+            {
+                this.transmitter.Flush(timeout);
+            }
+            else
+            {
+                throw new ObjectDisposedException("InMemoryChannel", "The telemetry channel has been disposed.");
```

Comment:
`Dispose` is just a method, how it's implemented is a design choice that needs to be tailored to a specific scenario. Not all classes in the .NET framework throw `ObjectDisposedException` and the `InMemoryChannel.Flush` shouldn't either. The only reason it's required in this PR is because the `InMemoryChannel.Dispose` now clears the `InMemoryChannel.transmitter` field. 

This is a breaking change that doesn't add user value. Please undo the `InMemoryChannel` changes.